### ISSUE TITLE
Update resource

### ISSUE THREAD

**Issue by alejo8591** [cmauroie/control#1]
``` javascript
.factory('AccessService_Factory', function($resource){
    return $resource('http://199.180.186.183/rest/access/initialData',{ },{
    //return $resource('http://192.168.34.21/Modipay/rest/access/initialData', { }, {
        update: {

            method:'GET',     
            headers:{
        'Content-Type':'application/json','Authorization':'Basic MzcyNDI4OnZhbGxlMTIzNA=='}
            }
 });
});
```

Change with:

``` javascript

.factory('AccessService_Factory', function($resource){
    return $resource('http://199.180.186.183/rest/access/initialData',{ },{
    //return $resource('http://192.168.34.21/Modipay/rest/access/initialData', { }, {
        get: {
            method:'GET',     
            headers:{
                        Content-Type':'application/json','Authorization':'Basic MzcyNDI4OnZhbGxlMTIzNA==' 
                      }
            }
 });
});
```

**Comment by cmauroie** [comment]
Hola Alejo 
Hice el cambio pero la respuesta sigue igual, y estuve buscando sobre el error que me sale en consola 
" No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'http://localhost:8100' is therefore not allowed access. The response had HTTP status code 401."
y me llevo a un tema llamado CORS, el cual verifica si la solicitud AJAX vienen del mismo dominio, si no es así no cargara los datos por limitaciones de seguridad, y pienso que debe ser el bug que se esta presentando.
en medio de la busca encontré que adicionando estas lineas a la cabecera pero tampoco me funciono.
'Access-Control-Allow-Headers':'Origin, X-Requested-With, Content-Type, Accept, Authorization','Access-Control-Allow-Origin':'Context.Request.Headers["Origin"]','Access-Control-Allow-Methods':'GET, POST PUT, DELETE, OPTIONS','Access-Control-Allow-Credentials':'true',

Carlos Mauricio Idárraga Espitia
3122611700Ingeniero Electrónico 

Date: Wed, 10 Jun 2015 11:13:40 -0700

**Comment by alejo8591** [comment]
Las cabeceras van el `backend` y deben añadirse según el lenguaje (Java, C#, Python, etc...) o el `Framework` de desarrollo. Para el ejemplo creo que tienes las de `JS`

**Comment by cmauroie** [comment]
http://blog.ionic.io/handling-cors-issues-in-ionic/
Alejo encontré este blog y al parecer se esta haciendo una verificación de los dominios y no puedo consumir este servicio,entiendo que como el dominio de la petición no coincide con el dominio de origen en mi caso localhost, entonces se activa la verificación CORS. ¿Será que no podre consumir este servicio?.    

Carlos Mauricio Idárraga Espitia
3122611700Ingeniero Electrónico 

Date: Wed, 10 Jun 2015 12:19:37 -0700

**Comment by cmauroie** [comment]
http://blog.ionic.io/handling-cors-issues-in-ionic/
Alejo encontré este blog y al parecer se esta haciendo una verificación de los dominios y no puedo consumir este servicio,entiendo que como el dominio de la petición no coincide con el dominio de origen en mi caso localhost, entonces se activa la verificación CORS. ¿Será que no podre consumir este servicio?.    

Carlos Mauricio Idárraga Espitia
3122611700Ingeniero Electrónico 

Date: Wed, 10 Jun 2015 12:19:37 -0700

**Comment by alejo8591** [comment]
Tienes que decirle a los desarrolladores de la aplicación de `backend` que habiliten las peticiones de otros demonios que son los mecanismos de configurar la respuesta en las cabeceras, ejemplo:

``` javascript
               /*
                * Set de cabeceras para poder permitir acceso
                * de peticiones AJAX de un request externo
                * en este caso la App
                * */
               res.header('Access-Control-Allow-Origin', '*');

               res.header('Access-Control-Allow-Methods', 'POST, PUT');

               res.header('Access-Control-Allow-Headers', 'Content-Type, Authorization, Content-Length, X-Requested-With, *');
```

Si no le hubiese agregado eso a la respuesta de las peticiones; específicamente a las cabeceras `HTTP` de la petición me saldría ese error `CORS`

**Comment by cmauroie** [comment]
Hola alejo como esta,
Alejo mira que configure las cabeceras del servidor para que permita el acceso ajax desde cualquier dominio pero el error sigue saliendo, en mi caso hice un servicio rest con cabeceras:

  System.out.println("processRequest center header: " + req.getHeader("Authorization"));                resp.setHeader("Access-Control-Allow-Origin","*");        resp.setHeader("Accept","application/json");        resp.setHeader("Cache-Control","no-cache, no-store, must-revalidate");        resp.setContentType("application/json");

******************************************+pero el error continua saliendo "Access-Control-Allow-Origin", entonces estaba pensando si en este caso es el servidor el que no me permite el acceso.

Carlos Mauricio Idárraga Espitia
3122611700Ingeniero Electrónico 

Date: Wed, 10 Jun 2015 15:12:48 -0700

**Comment by cmauroie** [comment]
Hola alejo, 
ya solucione el problema al parecer los navegadores restringen las solicitudes ajax desde un client javascript lo que hice,fue descargar un plugin de chrome llamado CORS y activar esta función y hasta aquí todo funciona bien,pero ahora la duda es como puedo hacer esto desde la app móvil.¿Se debe implementar un plugin de cordova?

Carlos Mauricio Idárraga Espitia
3122611700Ingeniero Electrónico 

From: mauricio198917@hotmail.com
To: reply@reply.github.com
Subject: RE: [control] Update resource (#1)
Date: Thu, 11 Jun 2015 16:48:42 -0500

Hola alejo como esta,
Alejo mira que configure las cabeceras del servidor para que permita el acceso ajax desde cualquier dominio pero el error sigue saliendo, en mi caso hice un servicio rest con cabeceras:

  System.out.println("processRequest center header: " + req.getHeader("Authorization"));                resp.setHeader("Access-Control-Allow-Origin","*");        resp.setHeader("Accept","application/json");        resp.setHeader("Cache-Control","no-cache, no-store, must-revalidate");        resp.setContentType("application/json");

******************************************+pero el error continua saliendo "Access-Control-Allow-Origin", entonces estaba pensando si en este caso es el servidor el que no me permite el acceso.

Carlos Mauricio Idárraga Espitia
3122611700Ingeniero Electrónico 

Date: Wed, 10 Jun 2015 15:12:48 -0700

**Comment by alejo8591** [comment]
No, no necesitas ningún plugin, desde la app puedes llamar el servicio.

**Comment by cmauroie** [comment]
Hola Alejo como estuvo el fin de semana,
Alejo le escribo para aclarar una pequeña duda, cuando quiero llamar una función desde un HTML lo hago de esta forma:
Cuando hacen click en el item Inicio se ejecuta la función "login()".
<ion-item menu-close ng-click="login()" class = "item-icon-left" >         <i class="icon ion-android-home balanced"></i>          Inicio        </ion-item>
Ahora mi duda es la siguiente si quiero llamar esta función desde un estado como lo puedo hacer.
.state('app.login', {    url: "/login",    views: {      'menuContent': {        templateUrl: "templates/login.html",       controller: 'AppCtrl'      }    }  })

Ya que inicie un proyecto como:
ionic start nameApp sidemenu
y quiero que el template de login me resalte antes de iniciar las opciones de la app para validar credenciales y así continuar, lo hice llamando el template y el controlado en el estado pero no se oculta los menus de la App, como puedo hacer para llamar la función login(); y activar el modal.show().

 $scope.login = function() {    $scope.modal.show();  };. 
.controller('AppCtrl', function($scope, $ionicModal, $timeout) {    // With the new view caching in Ionic, Controllers are only called  // when they are recreated or on app start, instead of every page change.  // To listen for when this page is active (for example, to refresh data),  // listen for the $ionicView.enter event:  //$scope.$on('$ionicView.enter', function(e) {  //});    // Form data for the login modal  $scope.loginData = {};
  // Create the login modal that we will use later  $ionicModal.fromTemplateUrl('templates/login.html', {    scope: $scope
  }).then(function(modal) {    $scope.modal = modal;  });
  // Triggered in the login modal to close it  $scope.closeLogin = function() {    $scope.modal.hide();  };
  // Open the login modal  $scope.login = function() {    $scope.modal.show();  };
  // Perform the login action when the user submits the login form  $scope.doLogin = function() {    console.log('Doing login', $scope.loginData);
    // Simulate a login delay. Remove this and replace with your login    // code if using a login system    $timeout(function() {      $scope.closeLogin();    }, 1000);  };})

Carlos Mauricio Idárraga Espitia
3122611700Ingeniero Electrónico 

Date: Thu, 11 Jun 2015 20:32:34 -0700
